woocommerce_integration/woocommerce_connector.py

```diff
@@ -22,80 +21,51 @@ def __init__(self, setup: dict):
             timeout=1000,
         )
 
-    def get_products(self, **kwargs):
-        filter_string = build_filter_string(kwargs)
+    def _request(self, method, endpoint, data=None, params=None, **kwargs):
+        response = self.woocommerce.__request(method, endpoint, data, params, **kwargs)
+
         try:
-            response = self.woocommerce.get(
-                f'products{f"?{filter_string}" if filter_string else ""}'
-            )
             response.raise_for_status()
-            return response.json()
-        except Exception:
+        except HTTPError as e:
             log_woocommerce_error(response)
-            raise
+            raise e
```

Comment:
Not sure if we really need this. For background jobs, this will likely result in duplicate error logs (`execute_job` also logs any errors). For direct user interaction, it might be fine to not log the error (the user sees the raised error anyway).